### DISCUSSION TITLE
How are you enabling dynamic typing in c?

### DISCUSSION THREAD

**Discussion by SaptakBhoumik** [edubart/nelua-lang#108] [Q&A] [answered]
Nelua is transcompiled to c so how are you enabling dynamic typing of lua in c? I was just curious to know

**Comment by edubart** [comment, marked as answer]
With type deduction, you can deduce variable types based on assignments or from a few type notations, this is currently how it works and this has no runtime costs, all is done at compile-time. If assignments have different types for the same variable, then you can use a tagged union for the types or an `any` type that can hold any value, but this time both have runtime costs, however this was not implemented yet.

**Reply by SaptakBhoumik** [reply]
Ok thanks
